### ISSUE TITLE
Reminder not shown when device was off or app force-closed at due time

### ISSUE THREAD

**Issue by felixwiemuth** [iSoron/uhabits#1808]
## Pre-submission checklist
- [X] I am submitting a bug report, not a feature request.
- [X] I have searched for similar issues, but did not find any matches.

## Description
Obviously, the app cannot show a reminder while the device is off or after the app was force-closed. But due reminders should be shown when the device is started or the app is launched.

I already mentioned this problem in #1509 where I implemented restoring of already shown notifications. Here we talk about reminders not shown yet. If the reminder cannot be shown at the planned time, it should be shown as soon as possible after it became due.

## Steps to reproduce
1. Create a habit with a reminder time in the future
2. Force-close the app or shut down the device
3. Wait until the reminder time has passed
4. Start the device and/or the app
The reminder is not shown.

## System information
not relevant

**Comment by Jineshpanchal** [comment]
Yes we also face the same issue, but it is very random in nature. sometime it works and sometime it doesn't. also based on android version it behaves diffrently.
